pacta_01.R

```diff
@@ -7,37 +7,62 @@ suppressPackageStartupMessages({
   library(jsonlite)
 })
 
+# defaulting to WARN to maintain current (silent) behavior.
+logger::log_threshold(Sys.getenv("LOG_LEVEL", "WARN"))
+logger::log_formatter(logger::formatter_glue)
 
 # -------------------------------------------------------------------------
 
+logger::log_info("Starting portfolio audit")
+
+logger::log_trace("Determining configuration file path")
 cfg_path <- commandArgs(trailingOnly = TRUE)
-if (length(cfg_path) == 0 || cfg_path == "") { cfg_path <- "input_dir/default_config.json" }
+if (length(cfg_path) == 0 || cfg_path == "") {
+  logger::log_warn("No configuration file specified, using default")
+  cfg_path <- "input_dir/default_config.json"
+}
+logger::log_debug("Loading configuration from file: \"{cfg_path}\".")
```

Comment:
NIT: Can you explain what `log_debug` does? Just something that triggers when you set `LOG_LEVEL=DEBUG` i suppose?

Reply:
`logger::log_debug()` emits a `DEBUG` level logging message. `logger` follows the `log4j` hierarchy of naming scheme (generally), of `TRACE` -> `DEBUG` -> `INFO` -> `SUCCESS` -> `WARN` -> `ERROR` -> `FATAL`, where setting a log level allows all "more important" messages also through. 

https://github.com/RMI-PACTA/workflow.pacta/blob/0a860d4a52f2a90370b34e3d08b17c31988d282b/pacta_01.R#L10-L12

These lines are where we're actually setting the log level (and log formatter, which tells our code to use `glue` strings for variable interpolation). So if the `LOG_LEVEL` envvar is not set, then we would only see messages from `log_warn()`, `log_error()` and `log_fatal()` emitted (`log_info()` and `log_debug()` would be silent. 

* Quick table explaining what log levels are active: https://stackoverflow.com/a/37167120

Reply:
Thank you! Sounds great. 

My final point, I really like that the logging occurs entirely on the root level `*.R` script in the `workflow.*` repo. 
My (and I think @cjyetman 's) worry about logging, is that they are likely to fall out of sync and cause more confusion than help should we forget to update them (similar to comments). 

But I think this is mitigated by ensuring they all appear up front in the `main` script. Especially since, in principle, these scripts should be a series of simple function calls. 

It may even incentivize us to keep the `main` script simple

Reply:
Hmm. I'm inclined to go the other way, and keep the logs in the function where they're called ([example](https://github.com/RMI-PACTA/workflow.factset/blob/main/R/get_entity_ids.R) from `workflow.factset`). That's still in the `workflow` repo, but more closely ties the log emitter to the thing that it's logging.

Unpopular opinion: I'd like to see a lot more `DEBUG` and `TRACE` (and maybe some `INFO`) logs in our packages, so that we can get useful diagnostics regardless of if it's called from `workflow.*` or not.

Reply:
Gotcha, then that's a Tech Review topic, as it pretty fundamentally goes against most of @cjyetman and I's "unspoken but pretty consistently implemented" practices and style. And is a pretty huge departure from how we have worked and developed until now.

Reply:
For the record, I am also pretty strongly against that opinion (but as usual may be convinced otherwise)